File: common/src/main/java/com/pluxity/cctv/dto/CctvResponse.java
package com.pluxity.cctv.dto;

import com.pluxity.cctv.category.dto.CctvCategoryResponse;
import com.pluxity.feature.dto.FeatureResponse;
import lombok.Builder;

@Builder
public record CctvResponse(
        String id,
        String name,
        String url,
        FeatureResponse feature,
        CctvCategoryResponse cctvCategory) {}

File: gs/src/main/kotlin/com/pluxity/cctv/dto/CctvResponse.kt
package com.pluxity.cctv.dto

import com.pluxity.cctv.entity.Cctv
import com.pluxity.feature.dto.FeatureResponse
import com.pluxity.feature.dto.toFeatureResponse

data class CctvResponse(
    val id: String,
    val name: String,
    val viewUrl: String? = null,
    val url: String?,
    val feature: FeatureResponse?,
)

fun Cctv.toCctvResponse(viewUrl: String) =
    CctvResponse(
        id = this.id,
        name = this.name,
        url = this.url,
        viewUrl = this.mtxName?.let { "$viewUrl/${this.mtxName}" },
        feature = this.feature?.toFeatureResponse(),
    )
